receiver/awscontainerinsightreceiver/internal/neuron/pod_attribute_decorator.go

```diff
@@ -0,0 +1,131 @@
+package neuron
+
+import (
+	"context"
+	"strconv"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	neuronHardwareInfoKey       = "neuron_hardware"
+	neuronCorePerDeviceKey      = "neuroncore_per_device_count"
+	neuronCoreAttributeKey      = "neuroncore"
+	neuronDeviceAttributeKey    = "neuron_device_index"
+	neuronCoreResourceName      = "aws.amazon.com/neuroncore"
+	neuronDeviceResourceName    = "aws.amazon.com/neurondevice"
+	neuronDeviceResourceNameAlt = "aws.amazon.com/neuron"
+)
+
+type PodAttributesDecoratorConsumer struct {
+	NextConsumer      consumer.Metrics
+	PodResourcesStore *stores.PodResourcesStore // replace with podResourcesApi
+	Logger            *zap.Logger
+}
+
+func (pdc *PodAttributesDecoratorConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: true,
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	md = pdc.neuronMetricsProcess(md)
+	return pdc.NextConsumer.ConsumeMetrics(ctx, md)
+}
+
+func (pdc *PodAttributesDecoratorConsumer) neuronMetricsProcess(md pmetric.Metrics) pmetric.Metrics {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rs := rms.At(i)
+		ilms := rs.ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ils := ilms.At(j)
+			metrics := ils.Metrics()
+
+			neuronHardwareInfo := pmetric.Metric{}
```

Comment:
`var neuronHardwareInfo pmetric.Metric`

Also, need to handle the case where we don't find the expected neuron_hardware metric.